src/Compilers/CSharp/Test/Emit2/Semantics/RefReadonlyParameterTests.cs

```diff
@@ -1191,13 +1191,229 @@ class C
+
+        // Consumer can use the API.
+        var source2 = """
+            public class D
+            {
+                public unsafe void M()
+                {
+                    new C().M(&F);
+                }
+                static void F(ref readonly int x) => System.Console.Write("F" + x);
+            }
+            """;
+        var comp2 = CreateCompilation(source2, new[] { comp1v1Ref }, options: TestOptions.UnsafeReleaseDll).VerifyDiagnostics();
+        var comp2Ref = comp2.EmitToImageReference();
+
+        var source3 = """
+            try
+            {
+                new D().M();
+            }
+            catch (System.MissingMethodException e)
+            {
+                System.Console.Write(e.Message.Contains("Void C.M(Void (Int32 ByRef))"));
+            }
+            """;
+        CompileAndVerify(source3, new[] { comp1v1Ref, comp2Ref }, expectedOutput: "F123").VerifyDiagnostics();
+
+        // .NET runtime declares the attribute.
+        var source4 = """
+            namespace System.Runtime.CompilerServices
+            {
+                [AttributeUsage(AttributeTargets.Parameter, AllowMultiple = false, Inherited = false)]
+                public sealed class RequiresLocationAttribute : Attribute
+                {
+                }
+            }
+            """;
+        var comp4 = CreateCompilation(source4).VerifyDiagnostics();
+        var comp4Ref = comp4.EmitToImageReference();
+
+        // Library is recompiled against the newest runtime.
+        var comp1v2 = CreateCompilation(source1, new[] { comp4Ref }, assemblyName: "Assembly1", options: TestOptions.UnsafeReleaseDll);
+        comp1v2.VerifyDiagnostics();
+        var comp1v2Ref = comp1v2.EmitToImageReference();
+
+        // That breaks the consumer.
+        CompileAndVerify(source3, new[] { comp1v2Ref, comp2Ref, comp4Ref }, expectedOutput: "True").VerifyDiagnostics();
+
+        // Unless the library adds type forwarding.
+        var source5 = """
+            using System.Runtime.CompilerServices;
+            [assembly: TypeForwardedToAttribute(typeof(RequiresLocationAttribute))]
+            """;
+        var comp1v3 = CreateCompilation(new[] { source1, source5 }, new[] { comp4Ref }, assemblyName: "Assembly1", options: TestOptions.UnsafeReleaseDll);
+        comp1v3.VerifyDiagnostics();
+        var comp1v3Ref = comp1v3.EmitToImageReference();
+        CompileAndVerify(source3, new[] { comp1v3Ref, comp2Ref, comp4Ref }, expectedOutput: "F123").VerifyDiagnostics();
+
+        // Or keeps the manual attribute definition.
+        var comp1v4 = CreateCompilation(new[] { source1, RequiresLocationAttributeDefinition }, new[] { comp4Ref }, assemblyName: "Assembly1", options: TestOptions.UnsafeReleaseDll);
+        comp1v4.VerifyDiagnostics();
+        var comp1v4Ref = comp1v4.EmitToImageReference();
+        CompileAndVerify(source3, new[] { comp1v4Ref, comp2Ref, comp4Ref }, expectedOutput: "F123").VerifyDiagnostics();
```

Comment:
Should this one check that the assembly name emitted in the `modopt` is what's expected?